smdebug/xgboost/hook.py

```diff
@@ -219,6 +228,18 @@ def write_hyperparameters(self, env: CallbackEnv):
         for param_name, param_value in self.hyperparameters.items():
             self._save_for_tensor("hyperparameters/{}".format(param_name), param_value)
 
+    def write_losses(self, env: CallbackEnv):
+        # Get loss metric at the current boosting round
+        # loss metrics will have already been saved by write_metrics if the metrics collection is included.
+        if self._is_collection_being_saved_for_step(CollectionKeys.METRICS):
+            return
+        else:
+            evaluation_metrics = env.evaluation_result_list
+            increasing_metric_names = _filter_increasing_metric_names(evaluation_metrics)
+            for metric_name, metric_data in evaluation_metrics:
```

Comment:
This could be slightly refactored to be more simple. 

```
evaluation_metrics = env.evaluation_result_list
p = re.compile("^[a-zA-z]+-(auc|map)$") # regex for matching increasing metrics
for metric_name, metric_data in evaluation_metrics:
    if p.match(metric_name):
        continue
    self._save_for_tensor(metric_name, metric_data)
    
```
Thoughts?